03-devops-overview.Rmd

```diff
@@ -0,0 +1,183 @@
+# DevOps - The Basics
+
+## Background
+
+So far we have only looked at Git in the context of local projects and command line software. This alone doesn't make collaborating on code easy. 
+
+What is needed is a central platform where people can request for changes to be made to the code base and update their code based on changes by others.  
+
+This concept in Git is known as [remotes](https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes) or remote repositories.
+The following are examples of interactive services that allow you to manage Git projects effectively:
+
+- [GitHub](github.com)
+- [Bitbucket](https://bitbucket.org)
+- [GitLab](gitlab.com/)
+- [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/).
+
+We are using **Azure DevOps** as this is the Department's tool of choice for internal version control. We would use platforms such as GitHub for our more publicly facing projects (such as this book).
+
+To work with internal DevOps Repos you will need an account with Basic access level.
+There is a form to sort out DevOps account levels and access on the IT service portal - search for DevOps.
+You will also need someone to give you permission to see the DevOps project that you're working on.
+For information about how to do this, pleae read on.
+If you need a new project area on DevOps, use the same IT service portal form as for account access.
+
+## Theory 
+
+At a high level when you are using DevOps (or any type of remote) you are creating a central copy of your code on a server (the DevOps site). 
+
+The aim of any project is to get your final complete code on to the main branch of the server copy. Analysts will still make and run all their changes locally but will then use the concepts of *pushing*, *pulling* and *pull requesting* to get the DevOps version up to date.
+
+Having this central repository and the approval processes that DevOps provides means that multiple analysts can work on a project at the same time and systematically merge all their changes into the final version after code review.
+
+The following diagram illustrates the concept of server and local copies. 
+
+![](www/distributed_version_control.png)
+
+## Creating a DevOps repository for a new project
+
+To set up a repository on DevOps you need to:
+
+1. Open the DevOps website.
```

Comment:
A link to DevOps website?

Reply:
Sadly not, I think that URL should be kept internal